File: bindings/python/tests/Test_MultiIndex.py
# MultiIndex test

import mpart
import unittest

class TestStringMethods(unittest.TestCase):


    def test_1(self):
        
        length = 4
        multi = mpart.MultiIndex(length)
        assert len(multi) == length
        assert multi.count_nonzero() == 0
        assert multi[0] == 0
        assert multi.max() == 0
        assert multi.sum() == 0

        multi[0] = 4
        assert multi[0] == 4
        assert multi.max() == 4
        assert multi.sum() == 4

        multi[1] = 3
        assert multi[1] == 3
        assert multi.max() == 4
        assert multi.sum() == 4 + 3

    def test_2(self):
        
        length = 4
        val = 2
        multi = mpart.MultiIndex(length,val)
        assert len(multi) == length
        assert multi.count_nonzero() == length
        assert multi[0] == val
        assert multi.max() == val
        assert multi.sum() == length*val


    def test_3(self):
        
        idx = [2, 3, 0, 0, 100]
        multi = mpart.MultiIndex(idx)
        assert len(multi) == len(idx)
        assert multi.count_nonzero() == 3
        assert multi[0] == idx[0]
        assert multi.max() == 100
        assert multi.sum() == 105
        assert multi.tolist() == idx


if __name__ == '__main__':
    unittest.main()

File: bindings/julia/tests/Test_MultiIndex.jl
using Test
using .MParT

function test_sum()
    idx = MParT.MultiIndex(30,2)
    @test sum(idx) == 60
    idx[1] = 3
    @test sum(idx) == 61
end

function test_max()
    idx = MParT.MultiIndex(30,2)
    @test maximum(idx) == 2
    idx[1] = 100
    @test maximum(idx) == 100
end

function test_count_nonzero()
    idx = MParT.MultiIndex(30)
    @test count_nonzero(idx) == 0
    idx[4] = 1
    idx[27] = 3
    @test count_nonzero(idx) == 2
end

@testset verbose=true "MParT" begin
    @testset verbose=true "MultiIndex" begin
        test_sum()
        test_max()
        test_count_nonzero()
    end
end